### DISCUSSION TITLE
Getting the Schema of a Collection

### DISCUSSION THREAD

**Discussion by RayyanNafees** [pocketbase/pocketbase#3950] [Q&A] [answered]
What is the process (Extend with JS/GO or Web APIs or otherwise) to get the schema of a particular collection

## For Eg:
```js
const frontmatterTypes = pb.collection('blogs').schema()
console.log(frontmatterTypes)
```

## Result
```js
[
            {
                name: 'status',
                type: 'bool',
            },
        
            {
                name: 'title',
                type: 'text',
            },
  ]
```

**Comment by ganigeorgiev** [comment, marked as answer]
The question is too vague and it is not clear what you are asking and what you are trying to do.

In general, the collection schema is part of the collection model.

When using the web APIs it will be returned under the `schema` key. For example, see the sample response in https://pocketbase.io/docs/api-collections/#view-collection (collections web APIs require admin authentication).

For accessing the schema fields with Go, it could be done via the `models.Collection.Schema` field. All `Schema` field methods and fields could be explored in https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.20.0/models/schema#Schema.

For JS is similar since the JS model is a translation of the Go one. The documentation for the collection can be found in https://pocketbase.io/jsvm/interfaces/models.Collection.html and https://pocketbase.io/jsvm/interfaces/schema.Schema.html.
